reactnative_mobile/react_app_todo_list/App.tsx

```diff
@@ -30,27 +54,89 @@ export default function App() {
     appState.current = nextAppState;
   };
 
-  useEffect(() => {
-    const subscription = AppState.addEventListener(
-      "change",
-      handleAppStateChange
-    );
-    return () => {
-      subscription.remove();
-    };
-  }, []);
-
-  const handleAddTask = async (text: string) => {
+  const handleAddTask = async (modelObj: TODOModel) => {
     Keyboard.dismiss();
-    setTaskItems([...taskItems, task]);
-    setTask(null);
+    setTask("");
     Statsig.logEvent(TODO_CREATED);
+    addTodoInList(modelObj);
+    fetchTodoList();
+  };
+
+  const addTodoInList = async (modelObj: TODOModel) => {
+    fetch(baseTodoUrl, {
```

Comment:
These should live in their own file. Maybe use a hook like `const {addTodoItem, fetchTodoItems, deleteTodoItem} = useTodoService();`